lxd/instance_put.go

```diff
@@ -202,6 +203,15 @@ func instanceSnapRestore(s *state.State, projectName string, name string, snap s
 		}
 	}
 
+	// Generate a new UUID for the instance when restoring it.
+	// We will later use this new instance UUID as a VM Generation ID
+	// in order to differentiate the two instances (before and after snapshotting)
```

Comment:
Not "before and after snapshotting", but "before and after snapshot restore that changes that instance's place in time (moving it backwards)"